lib/basedriver/helpers.js

```diff
@@ -321,7 +321,12 @@ async function unzipApp (zipPath, dstRoot, supportedAppExtensions) {
   const tmpRoot = await tempDir.openDir();
   try {
     logger.debug(`Unzipping '${zipPath}'`);
-    await zip.extractAllTo(zipPath, tmpRoot);
+    // zip.extractAllTo could not unzip Unicode filenames properly by Xcode
+    // because of https://github.com/appium/appium/issues/14100
+    // so here uses unzipper which can handle the case
+    await unzipper.Open.file(zipPath)
+      .then(d => d.extract({path: tmpRoot, concurrency: 5}));
```

Comment:
we don't need `.then` since Open also returns a promise

Reply:
```
const directory = await OpenZip.file(...);
await directory.extract(...);
```

Reply:
it makes sense to set concurrency equal to the cpu cores count (`os.cpus().length`)

Reply:
yey, it is I wanted to find... > os.cpus().length